tests/unit/api/main.py

```diff
@@ -132,6 +132,54 @@ def test_duplicate_project_create(self):
         with pytest.raises(api.DuplicateError):
             api.project_create('acme-corp')
 
+class TestProjectAddDeleteNetwork:
```

Comment:
Two options: (1) use `additional_db`. If you need some extra objects in there to make these tests work, add them. (2) add a `setUp` function to the class that creates the projects and a few networks that are in an initial state that you can just use.

General rule of thumb, try not to have API calls besides the one you're testing in the test. Also, actually state explicitly what property the test is checking

If you have tests with incompatible requirements for the database comments, try to split them into (large) groups of compatible requirements, and split them into classes based on that (with their own setup function).

Also, make sure you're explicitly stating what the tests are checking -- what about these api calls are you testing? (looks like checking that side effects of the calls actually take place).